zoninator.php

```diff
@@ -650,9 +655,22 @@ function zone_admin_recent_posts_dropdown( $zone_id ) {
 			'suppress_filters' => false,
 		) );
 
+		/**
+		 * Filters the value of recent posts array before it is retrieved.
+		 *
+		 * Passing a non-empty array to the filter will short-circuit retrieving
+		 * the recent posts via get_posts, using the passed array of posts instead.
+		 *
+		 * @param array $recent_posts        Array of WP_Posts objects to use. Default empty
+		 *                                   array to retrieve the posts via get_pots.
+		 * @param array $args                Array of get_posts parameters.
+		 * @param int|string|object $zone_id Zone ID being processed.
+		 */
+		$recent_posts = apply_filters( 'zoninator_recent_posts', array(), $args, $zone_id );
```

Comment:
A think a more common pattern for an approach like this is to label the filter as `pre` (e.g. `pre_option_*`:

```
$recent_posts = apply_filters( 'zoninator_pre_recent_posts', false, $args, $zone_id );
if ( false === $recent_posts ) {
    $recent_posts = get_posts( $args );
}
```

`zoninator_recent_posts` would generally make think that it allows me to filter the posts after they have been fetched.